File: framework/languages/java/samples/java/DeliveryDispatch/Server/Session/src/main/java/systems/zlink/samples/deliverydispatch/server/session/sessions/handlers/DeliveryStatusFanoutHandler.java
package systems.zlink.samples.deliverydispatch.server.session.sessions.handlers;

import systems.zlink.framework.channels.ZLinkPublishContext;
import systems.zlink.framework.channels.ZLinkPublishHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.deliverydispatch.server.session.sessions.CustomerSessionDirectory;
import systems.zlink.samples.deliverydispatch.shared.contracts.Messages;

@ZLinkHandlerGroup("status")
public final class DeliveryStatusFanoutHandler
    implements ZLinkPublishHandler<Messages.DeliveryStatusNotify> {
    private final CustomerSessionDirectory sessions;

    public DeliveryStatusFanoutHandler(CustomerSessionDirectory sessions) {
        this.sessions = sessions;
    }

    @Override
    public void handle(Messages.DeliveryStatusNotify message, ZLinkPublishContext context) {
        sessions.publish(message);
    }
}

File: framework/languages/java/samples/kotlin/DeliveryDispatch/Server/Session/src/main/kotlin/systems/zlink/samples/kotlin/deliverydispatch/server/session/sessions/handlers/DeliveryStatusFanoutHandler.kt
package systems.zlink.samples.kotlin.deliverydispatch.server.session.sessions.handlers

import systems.zlink.framework.channels.ZLinkPublishContext
import systems.zlink.framework.kotlin.ZLinkSuspendingPublishHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.samples.kotlin.deliverydispatch.server.session.sessions.CustomerSessionDirectory
import systems.zlink.samples.kotlin.deliverydispatch.shared.contracts.DeliveryStatusNotify

@ZLinkHandlerGroup("status")
class DeliveryStatusFanoutHandler(
    private val sessions: CustomerSessionDirectory,
) : ZLinkSuspendingPublishHandler<DeliveryStatusNotify> {
    override suspend fun handle(message: DeliveryStatusNotify, context: ZLinkPublishContext) = run {
        sessions.publish(message)
    }
}
